feat: add git workflow integration tools

JimStenstrom opened the pull request:
## Description

Add 4 git workflow tools for enhanced version control integration (Phase 1 of issue #98):

- **git_status_enhanced** - Enhanced git status with stash info, recent commits, and actionable suggestions
- **git_branch_suggest** - Intelligent branch name suggestions based on work type and description
- **git_smart_commit** - Conventional commit message generation from staged changes
- **git_create_pr** - PR template generation with summary, test plan, and labels

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

## Testing

- [ ] Tested with Ollama
- [ ] Tested with OpenRouter
- [x] Tested with OpenAI-compatible API
- [ ] Tested MCP integration (if applicable)

## Checklist

- [x] Code follows project style guidelines
- [x] Self-review completed
- [ ] Documentation updated (if needed)
- [x] No breaking changes (or clearly documented)

---

**Additional Details:**

- 40 unit tests pass (`pnpm test:all`)
- Build succeeds (`pnpm build`)
- Tested in interactive and non-interactive modes
- Follows `NanocoderToolExport` pattern

Closes #98

JimStenstrom commented:
Ready for a new review.

will-lamerton commented on source/tools/git/git-smart-commit.tsx:
When running I can't appear to see the chosen commit message? Let me know if I'm misunderstanding how this tool works but if it's going to commit it would be good to see the message it's going for so either the user can accept or amend.

<img width="388" height="476" alt="Screenshot 2025-12-27 at 23 15 20" src="https://github.com/user-attachments/assets/9fb7b56e-9cd4-4c74-a230-d4198589fdda" />

will-lamerton commented on source/tools/git/git-branch-suggest.tsx:
```diff
@@ -0,0 +1,442 @@
+
+		const messageContent = (
+			<Box flexDirection="column">
+				<Text color={colors.tool}>git branch_suggest</Text>
```
This needs to say "git_branch_suggest" to be congruent with other tool formatters.

will-lamerton commented on source/tools/git/git-create-pr.tsx:
```diff
@@ -0,0 +1,503 @@
+
+		const messageContent = (
+			<Box flexDirection="column">
+				<Text color={colors.tool}>git create_pr</Text>
```
This needs to say "git_status_enhanced" to be congruent with other tool formatters.

will-lamerton reviewed (changes requested):
This is really cool to have in the works! Let me know if the comments are okay.

JimStenstrom commented:
## Addressed Review Feedback

Thanks for the detailed review! I've pushed fixes for the following:

### Fixed:
1. **Formatter names** - Changed from `git smart_commit` to `git_smart_commit` (and similar for all 4 tools)
2. **Show commit message** - `git_smart_commit` formatter now displays the full commit message so users can see it before accepting
3. **Show PR description** - `git_create_pr` formatter now shows a preview of the generated description
4. **Branch validation** - `generateAlternatives` now handles special-char-only descriptions with a fallback

### Already addressed in previous commits:
- `dryRun` default behavior (correctly defaults to true via `args.dryRun !== false`)
- `needsApproval` for destructive commit operations
- String escaping in commit message
- Error handling in `execGit` (rejects on non-zero exit)
- `parseInt` NaN handling
- Exact file path matching (not substring)

Ready for re-review!
